examples/mnist_dcgan.py

```diff
@@ -0,0 +1,136 @@
+"""
+Trains a DCGAN on the MNIST dataset.
+"""
+
+from __future__ import print_function, division
+import numpy as np
+from keras.datasets import mnist
+from keras.models import Sequential
+from keras.layers import Dense, Activation, Flatten, LeakyReLU, MinibatchDiscrimination
+from keras.layers import Convolution2D, UpSampling2D, Reshape
+from keras.optimizers import Adam
+try:
+    import matplotlib.pyplot as plt
+    plotting_enabled = True
+except ImportError:
+    plotting_enabled = False
+
+np.random.seed(42)  # for reproducibility
+
+
+def set_trainable(net, boolean):
+    net.trainable = boolean
```

Comment:
If it is then it is. I based the use of a function setting layer.trainable off of someone else's example, and some issue discussion. If setting net.trainable is enough, I'll remove the function and change the function calls to this line.